### PR TITLE
Added tests to geometry classes

### PR DESCRIPTION
**Pull request by molhamkhalil**
* rotation tests are not done yet, hence they are commented

### CODE REVIEW

**Comment by molhamkhalil** [comment]
* FloatTransform2D testing is done,
* setRotation error was a fault of the tester (me).

File: test/geometry/FloatTransform2DTest.ooc [hunk 1]
```diff
@@ -10,12 +10,15 @@ use unit
 use geometry
 
 FloatTransform2DTest: class extends Fixture {
+	PI := 3.14159265f
```
Comment (thomasfanell) [inline test/geometry/FloatTransform2DTest.ooc]: `pi` is already defined. Do a `import math` and use `Float pi` to access the constant.

File: test/geometry/FloatTransform2DTest.ooc [hunk 2]
```diff
@@ -170,6 +173,248 @@ FloatTransform2DTest: class extends Fixture {
...
+			expect(transform c, is equal to(0.0f) within(this precision))
+			expect(transform d, is equal to(2.0f) within(this precision))
+			expect(transform e, is equal to(1.78885438f) within(this precision))
+			expect(transform f, is equal to(0.0f) within(this precision))
+			expect(transform g, is equal to(5.0f) within(this precision))
+			expect(transform h, is equal to(12.52198066f) within(this precision))
+			expect(transform i, is equal to(1.0f) within(this precision))
+		})
+		this add("skewX", func {
+			transform := this transform0 skewX(3.14159265/6)
```
Comment (thomasfanell) [inline test/geometry/FloatTransform2DTest.ooc]: `pi` is hard-coded here.

File: test/geometry/FloatTransform2DTest.ooc [hunk 3]
```diff
@@ -170,6 +173,248 @@ FloatTransform2DTest: class extends Fixture {
...
+			expect(transform c, is equal to(0.0f) within(this precision))
+			expect(transform d, is equal to(2.5f) within(this precision))
+			expect(transform e, is equal to(1.0f) within(this precision))
+			expect(transform f, is equal to(0.0f) within(this precision))
+			expect(transform g, is equal to(8.5f) within(this precision))
+			expect(transform h, is equal to(7.0f) within(this precision))
+			expect(transform i, is equal to(1.0f) within(this precision))
+		})
+		this add("skewY", func {
+			transform := this transform0 skewY(3.14159265/6)
```
Comment (thomasfanell) [inline test/geometry/FloatTransform2DTest.ooc]: and here...

File: test/geometry/FloatTransform2DTest.ooc [hunk 4]
```diff
@@ -170,6 +173,248 @@ FloatTransform2DTest: class extends Fixture {
...
+			expect(transform g, is equal to(5))
+			expect(transform h, is equal to(7))
+			expect(transform i, is equal to(1))
+		})
+		this add("toString", func {
+			string := FloatTransform2D new(3.123456789f, 1.123456789f, 0.12365f, -11.52416f, 0.0f, 1.9) toString()
+			expect(string, is equal to("3.123457, 1.123457, 0.000000\t0.123650, -11.524160, 0.000000\t0.000000, 1.900000, 1.000000\t"))
+		})
+		this add("createSkewingX", func {
+			skewingX := FloatTransform2D createSkewingX(4 * this PI / 3)
```
Comment (thomasfanell) [inline test/geometry/FloatTransform2DTest.ooc]: You are mixing Int with Float here. Prefer using `4.0f` and `3.0f` instead.

**Comment by molhamkhalil** [comment]
@niklasborwell peer review?

File: test/geometry/FloatTransform2DTest.ooc [hunk 5]
```diff
@@ -170,6 +172,248 @@ FloatTransform2DTest: class extends Fixture {
...
+			expect(transform1 isAffine, is true)
+			expect(transform2 isAffine, is true)
+			expect(transform3 isAffine, is true)
+			expect(transform4 isAffine, is true)
+			expect(transform5 isAffine, is true)
+			expect(transform6 isAffine, is equal to (!transform5 isAffine))
+		})
+		this add("create(translation, scale, rotation) -> This", func {
+			arr := FloatTransform2D create(transform0 translation, transform0 scaling, transform0 rotationZ)
+			expect(arr a, is equal to (2.56066017f) within (this precision))
```
Comment (fredrikbryntesson) [inline test/geometry/FloatTransform2DTest.ooc]: Why is it named arr?

File: test/geometry/FloatTransform3DTest.ooc [hunk 6]
```diff
@@ -296,6 +297,431 @@ FloatTransform3DTest: class extends Fixture {
...
+			resultedVectorList := transform0 transformAndProjectCorners(FloatBox2D new (1.0f, 2.0f, 3.0f, 4.0f), Float epsilon - (Float epsilon / 2))
+			expect(resultedVectorList[0] x, is equal to (17.00000083447f) within (this precision))
+			expect(resultedVectorList[0] y, is equal to (23.00000095368f) within (this precision))
+			expect(resultedVectorList[1] x, is equal to (33.00000083447f) within (this precision))
+			expect(resultedVectorList[1] y, is equal to (43.00000095367f) within (this precision))
+			expect(resultedVectorList[2] x, is equal to (30.00000083446f) within (this precision))
+			expect(resultedVectorList[2] y, is equal to (49.00000095367f) within (this precision))
+			expect(resultedVectorList[3] x, is equal to (14.00000083447f) within (this precision))
+			expect(resultedVectorList[3] y, is equal to (29.00000095367f) within (this precision))
+		})
```
Comment (fredrikbryntesson) [inline test/geometry/FloatTransform3DTest.ooc]: Remove all empty lines

Comment (molhamkhalil) [inline test/geometry/FloatTransform3DTest.ooc]: all empty lines removed

File: test/geometry/FloatTransform2DTest.ooc (hunk 5 above)
Comment (molhamkhalil) [inline test/geometry/FloatTransform2DTest.ooc]: arr changed to transform.

File: test/geometry/FloatTransform3DTest.ooc (hunk 6 above)
Comment (molhamkhalil) [inline test/geometry/FloatTransform3DTest.ooc]: all empty lines removed

**Pull request merged**